pennylane/ops/op_math/condition.py

```diff
@@ -220,53 +220,45 @@ def __call_capture_enabled(self, *args, **kwargs):
         cond_prim = _get_cond_qfunc_prim()
 
         elifs = (
-            (self.orig_elifs,)
+            [self.orig_elifs]
             if len(self.orig_elifs) > 0 and not isinstance(self.orig_elifs[0], tuple)
-            else self.orig_elifs
+            else list(self.orig_elifs)
         )
-
-        flat_fn = FlatFn(functools.partial(self.true_fn, **kwargs))
-        jaxpr_true = jax.make_jaxpr(flat_fn)(*args)
-        jaxpr_false = (
-            jax.make_jaxpr(functools.partial(self.otherwise_fn, **kwargs))(*args)
-            if self.otherwise_fn
-            else None
-        )
-
-        # We extract each condition (or predicate) from the elifs argument list
-        # since these are traced by JAX and are passed as positional arguments to the primitive
-        elifs_conditions = []
-        jaxpr_elifs = []
-
-        for pred, elif_fn in elifs:
-            elifs_conditions.append(pred)
-            jaxpr_elifs.append(jax.make_jaxpr(functools.partial(elif_fn, **kwargs))(*args))
-
-        conditions = [self.condition, *elifs_conditions, True]
-
-        jaxpr_branches = [jaxpr_true, *jaxpr_elifs, jaxpr_false]
-        jaxpr_consts = [jaxpr.consts if jaxpr is not None else () for jaxpr in jaxpr_branches]
-        jaxpr_branches = [j.jaxpr if j else None for j in jaxpr_branches]
-
-        # We need to flatten the constants since JAX does not allow
-        # to pass lists as positional arguments
-        consts_flat = [const for sublist in jaxpr_consts for const in sublist]
-        n_consts_per_branch = [len(consts) for consts in jaxpr_consts]
+        flat_true_fn = FlatFn(self.true_fn)
+        branches = [(self.preds[0], flat_true_fn), *elifs, (True, self.otherwise_fn)]
+
+        end_const_ind = len(branches)  # consts go after conditions, first cond at len(branches)
+        conditions = []
+        jaxpr_branches = []
+        consts = []
+        consts_slices = []
+
+        for pred, fn in branches:
+            conditions.append(pred)
+            if fn is None:
+                jaxpr_branches.append(None)
+                consts_slices.append(slice(0, 0))
+            else:
+                jaxpr = jax.make_jaxpr(functools.partial(fn, **kwargs))(*args)
+                jaxpr_branches.append(jaxpr.jaxpr)
+                consts_slices.append(slice(end_const_ind, end_const_ind + len(jaxpr.consts)))
+                consts += jaxpr.consts

```

Comment:
`extend` could be an alternative here, not sure whether it matters.

Reply:
I tend to find `+` and `+=` more readable.  One might be more efficient than the other, but it probably doesn't matter in this case.